packages/grid/_modules_/grid/components/cell/GridCell.tsx

```diff
@@ -56,6 +56,9 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
   const valueToRender = formattedValue || value;
   const cellRef = React.useRef<HTMLDivElement>(null);
   const apiRef = React.useContext(GridApiContext);
+  const currentFocusedCell = apiRef!.current.getState().keyboard.cell;
+  const isCellFocusable =
+    (currentFocusedCell && hasFocus) || (rowIndex === 0 && colIndex === 0 && !currentFocusedCell);
```

Comment:
Why not keeping is as simple as?

```diff
diff --git a/packages/grid/_modules_/grid/hooks/features/core/gridState.ts b/packages/grid/_modules_/grid/hooks/features/core/gridState.ts
index 831b5be5..51996e25 100644
--- a/packages/grid/_modules_/grid/hooks/features/core/gridState.ts
+++ b/packages/grid/_modules_/grid/hooks/features/core/gridState.ts
@@ -66,7 +66,7 @@ export const getInitialGridState: () => GridState = () => ({
   scrollBar: { hasScrollX: false, hasScrollY: false, scrollBarSize: { x: 0, y: 0 } },
   viewportSizes: { width: 0, height: 1 },
   sorting: getInitialGridSortingState(),
-  keyboard: { cell: null, columnHeader: null, isMultipleKeyPressed: false },
+  keyboard: { cell: { colIndex: 0, rowIndex: 0 }, columnHeader: null, isMultipleKeyPressed: false },
   selection: {},
   filter: getInitialGridFilterState(),
   columnMenu: { open: false },
```

Reply:
As far as I understand doing it like that would mean that by default the first cell is always focused. This state only has a value once an interaction with the keyboard happens. If you click on a cell that state should remain emply.

Reply:
AFAIK, it's what the spec ask for. The first cell should be, by default, the active item in the roving index, meaning tabbable and focusable. But it doesn't mean that it should get the focus, IMHO the focus in an effect is plain wrong.

Reply:
I think that it comes down to what should this state host?

I would personally advocate for the tabIndex value, not the activeElement. I think that the focus should be managed imperatively. I also think that the performance of they keyboard is far from great. It's really lagging, compared to ag-Grid. I think that the only way to solve this is by managing the tabIndex and focus at the DOM level, bypassing React. It's the only solution we found on the Autocomplete to make the keyboard handler 
feel instant.

Reply:
Alright, then how about I create a separate ticket focused on only refactoring the keyboard state because now this change will also impact the logic from https://github.com/mui-org/material-ui-x/pull/1289?

Reply:
Sounds good